test/handlers/URMCompoundManager.handler.sol

```diff
@@ -0,0 +1,161 @@
+
+    // Create rollback transaction arrays
+    _targets = new address[](_numTransactions);
+    _values = new uint256[](_numTransactions);
+    _calldatas = new bytes[](_numTransactions);
+
+    for (uint256 _i = 0; _i < _numTransactions; _i++) {
+      // Select a random target for each transaction
+      uint256 _targetIndex = bound(_rollbackFee + _i, 0, targets.length - 1);
+      FakeProtocolContract _target = targets[_targetIndex];
+
+      // Select a random selector for each transaction
+      uint256 _selectorIndex = bound(uint256(uint160(_rollbackGuardian)) + _i, 0, selectors.length - 1);
+      bytes4 _selector = selectors[_selectorIndex];
+
+      _targets[_i] = address(_target);
+      _values[_i] = 0;
+
+      // Encode the calldata based on the selector
+      if (_selector == FakeProtocolContract.setFee.selector) {
+        _calldatas[_i] = abi.encodeWithSelector(_selector, _rollbackFee);
+      } else {
+        _calldatas[_i] = abi.encodeWithSelector(_selector, _rollbackGuardian);
+      }
+    }
+  }
+
+  /// @notice Propose a rollback
+  /// @param _rollbackFee The new fee
+  /// @param _rollbackGuardian The new guardian
+  function propose(uint256 _rollbackFee, address _rollbackGuardian) public countCall("propose") {
+    // Get the rollback transactions
+    (address[] memory _targets, uint256[] memory _values, bytes[] memory _calldatas) =
+      _generateRollback(_rollbackFee, _rollbackGuardian);
+
+    // Get the rollback ID
+    uint256 _rollbackId = urm.getRollbackId(_targets, _values, _calldatas, DESCRIPTION);
+
+    // Only propose if it doesn't already exist
+    if (LibRollbackSet.contains(rollbackSet, _rollbackId)) {
+      // ASK: SHOULD WE TRACK THIS IN CALL SUMMARY?
+      revert("Rollback already exists");
+    }
+
+    vm.prank(admin);
+    urm.propose(_targets, _values, _calldatas, DESCRIPTION);
+
+    // Get the actual state from URM and store in RollbackSet
+    ProposalState _actualState = urm.state(_rollbackId);
+    LibRollbackSet.add(
+      rollbackSet,
+      RollbackProposal({
+        targets: _targets,
+        values: _values,
+        calldatas: _calldatas,
+        description: DESCRIPTION,
+        rollbackId: _rollbackId,
+        state: _actualState
+      })
+    );
+  }
+
+  /// @notice Queue a valid rollback proposal
+  /// @param _randomIndex Used to randomly select a queueable proposal
+  function queue(uint256 _randomIndex) public countCall("queue") {
+    // Get a random proposed proposal
+    RollbackProposal memory _randomProposal =
+      LibRollbackSet.randByState(rollbackSet, ProposalState.Pending, _randomIndex);
+
+    // Check if the proposal has not expired
+    if (block.timestamp > urm.getRollback(_randomProposal.rollbackId).queueExpiresAt) {
+      revert("Proposal has expired");
+    }
+
+    // Queue the proposal
+    vm.prank(guardian);
+    urm.queue(_randomProposal.targets, _randomProposal.values, _randomProposal.calldatas, _randomProposal.description);
```

Comment:
it makes sense to always have the guardian call this, but then, maybe sometimes someone else should try

Reply:
I can add another function `invalidCallerOnQueue` / `invalidCallerOnExecute` / `invalidCallerOnCancel`
to track those scenarios